### DISCUSSION TITLE
dynamic link - deep link parameters on android

### DISCUSSION THREAD

**Discussion by batmanashvili** [invertase/react-native-firebase#6598] [General]
I am using dynamic link in React Native, I have all the steps required for IOS and android setup. (dynamic link opens the app on both platform) but when I have deep link attached to it, on Android app is opened but deep link parameters are not passed. 

IOS works fine, deep link works correctly.

```      
"react-native": "0.67.4",
"@react-native-firebase/app": "12.7.4",
"@react-native-firebase/dynamic-links": "12.7.4",

device: Xiaomi mi 9,
browser: Xiaomi browser,

deeplink: https://sweeftdigital.page.link/transfer_tip?userId=3
```   
### deep link debug version

<img width="1068" alt="Screenshot 2022-10-09 at 16 25 20" src="https://user-images.githubusercontent.com/62855343/194756712-749ef71c-86ca-498b-bf42-fa990d50d34b.png">

### video

In this video there are two browser: Mi browser and chrome browser, 
1. Mi browser: when pressing dynamic link, it opens app directly without passing parameters.
2. chrome: when pressing dynamic link, it asks to open either in browser or open app, when you choose and app it passes parameters and deep link works.
 

https://user-images.githubusercontent.com/62855343/194759711-90720b17-f623-4fe0-aba6-450d1c8b2926.mp4

### AndroidManifest

``` 
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.tipapp">

  <uses-permission android:name="android.permission.INTERNET" />
  <uses-permission android:name="android.permission.CAMERA" />
  <uses-permission android:name="android.permission.RECORD_AUDIO" />
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
  <uses-permission android:name="android.permission.DOWNLOAD_WITHOUT_NOTIFICATION" />
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
  <uses-permission android:name="android.permission.VIBRATE" />
  <uses-feature android:name="android.hardware.camera" android:required="false" />
  <uses-feature android:name="android.hardware.camera.front" android:required="false" />

  <application android:name=".MainApplication"
    android:label="@string/app_name"
    android:icon="@mipmap/ic_launcher"
    android:allowBackup="false"
    android:usesCleartextTraffic="true"
    android:theme="@style/AppTheme">
    <activity android:name=".MainActivity"
      android:exported='true'
      android:label="@string/app_name"
      android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
      android:launchMode="singleTask"
      android:screenOrientation="portrait"
      android:windowSoftInputMode="adjustResize">

      <intent-filter android:autoVerify="true">
        <action android:name="android.intent.action.MAIN" />
        <category android:name="android.intent.category.LAUNCHER" />
        <action android:name="android.intent.action.DOWNLOAD_COMPLETE" />
      </intent-filter>

      <intent-filter >
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <!-- Accepts URIs that begin with "https://www.sweeftdigital.page.link/” -->
        <data android:scheme="https"
          android:host="sweeftdigital.page.link" />
        <!-- Accepts URIs that begin with "tipapp://” -->
        <data android:scheme="tipapp" />
      </intent-filter>

    </activity>
  </application>
</manifest>

```

### deep link handling with react navigation
``` 
const config = {
      screens: {
         webView: {
            path: '/transfer_tip',
            parse: {
               userId: (userId: any) => userId,
            },
         },
         authLoading: '*',
      },
   };

   const linking = {
      prefixes: ['tipapp://', 'https://sweeftdigital.page.link'],
      config: config,
   };
   
    <NavigationContainer
         linking={linking}
      ... 

```

Is there anything incorrect or is it firebase dynamic links normal behaviour?

**Comment by mikehardy** [comment]
When I see questions related to old versions, I stop and wait to hear confirmation you have updated and still have the same experience on current versions:

```
"@react-native-firebase/app": "12.7.4",
"@react-native-firebase/dynamic-links": "12.7.4",
```

Please let me know once you're on current stable, and assert that anything else you've written is still valid

**Reply by batmanashvili** [reply]
just upgraded to latest version, removed and reinstalled node_modules, cleared android folder and started app again.

<img width="805" alt="Screenshot 2022-10-09 at 20 17 27" src="https://user-images.githubusercontent.com/62855343/194767790-2782bd33-1371-481f-a101-030d3f1b024c.png">


### video
in this video you can see 2 behaviour: 
1. if you press the link and choose the browser deep link parameter are not passed.
2. if you press continue with app, deep link parameters are passed and everything works fine.

https://user-images.githubusercontent.com/62855343/194768065-c226ace5-9b04-400c-9bc4-36e50d0f0ace.mov


The problem is that 99% of time user presses "continue with browser" and sets it as default and deep link never works correctly.

**Reply by mikehardy** [reply]
You have 4 warnings in the debug view decoding the link, item 3 looks particularly interesting. https://firebase.google.com/docs/dynamic-links/operating-system-integrations#android_integration_flowchart
